WasmAllocator/src/main/java/org/apache/arrow/memory/WasmAllocationManager.java

```diff
@@ -0,0 +1,203 @@
+//package com.example;
+package org.apache.arrow.memory;
+import org.apache.arrow.flatbuf.Null;
+// import org.apache.arrow.memory.AllocationManager;
+// import org.apache.arrow.memory.ArrowBuf;
+// import org.apache.arrow.memory.BufferAllocator;
+// import org.apache.arrow.memory.ReferenceManager;
+import org.apache.arrow.memory.util.MemoryUtil;
+
+import org.wasmer.Instance;
+import org.wasmer.Memory;
+import org.wasmer.exports.Function;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import sun.misc.Unsafe;
+
+
+/**
+ * Allocation manager based on unsafe API.
+ */
+public final class WasmAllocationManager extends AllocationManager {
+
+    public static final AllocationManager.Factory FACTORY = new AllocationManager.Factory() {
+
+        @Override
+        public AllocationManager create(BufferAllocator accountingAllocator, long size) {
+            try {
+                return new WasmAllocationManager(accountingAllocator, size);
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+            return null;
+        }
+    
+        @Override
+        public ArrowBuf empty() {
+          return EMPTY_BUFFER;
+        }
+    };
+
+//   public static final AllocationManager.Factory FACTORY = new Factory() {
+//     @Override
+//     public AllocationManager create(BufferAllocator accountingAllocator, long size) {
+//       return new UnsafeAllocationManager(accountingAllocator, size);
+//     }
+
+//     @Override
+//     public ArrowBuf empty() {
+//       return EMPTY;
+//     }
+//   };
+    //static final UnsafeDirectLittleEndian EMPTY = INNER_ALLOCATOR.empty;    
+    static ArrowBuf EMPTY_BUFFER
```

Comment:
Q: did you look at where this is used? Would it be a problem that this isn't WASM memory? Maybe we need something like creating a zero sized array in WASM?